src/bin/sol-fbp-runner/Widget.js

```diff
@@ -0,0 +1,222 @@
+(function(window) {
+
+	'use strict'
+
+	function Widget(payload, key) {
+
+		//caching a secure reference to the instance
+		var instance = this;
+		//var selectedPort;
+
+		Widget.prototype.portsCount = 0;
```

Comment:
you shouldn't set a class property from an instance constructor. Maybe move this outside this function